src/main/java/com/spotlightspace/core/attachment/service/AttachmentService.java

```diff
@@ -2,43 +2,171 @@
+        Attachment newAttachment = attachmentRepository.save(Attachment.of(fileUrl, tableRole, tableId));
+        return GetAttachmentResponseDto.from(newAttachment);
+    }
+
+
+    // 이벤트 생성 시 첨부파일도 같이 생성할 때 접근
     @Transactional
     public void addAttachmentList(List<MultipartFile> files, Long id, TableRole tableRole) throws IOException {
         for (MultipartFile file : files) {
             saveAttachment(file, id, tableRole);
         }
     }
 
+    // 유저, review 생성 시 첨부파일도 같이 생성할 때 접근
     @Transactional
     public void addAttachment(MultipartFile file, Long id, TableRole tableRole) throws IOException {
         saveAttachment(file, id, tableRole);
     }
 
-    private void saveAttachment(MultipartFile file, Long id, TableRole tableRole) throws IOException {
+    // 해당 (유저, 리뷰, 이벤트) 삭제시 연관 삭제는 여기로
+    @Transactional
+    public void deleteAttachmentWithOtherTable(Long tableId, TableRole tableRole) {
+        // 해당 테이블 id와 tableRole이 일치하는 첨부파일 찾기
+        Attachment attachment = attachmentRepository.findByTargetIdAndTableRoleOrElseThrow(tableId, tableRole);
+        String fileName = attachment.getUrl().substring(attachment.getUrl().lastIndexOf("/") + 1);
+        amazonS3Client.deleteObject(bucket, fileName);
+        attachmentRepository.delete(attachment);
+    }
+
+    // 첨부파일만 접근해서 삭제할 때
+    @Transactional
+    public void deleteAttachment(Long attachementId, Long tableId, TableRole tableRole, AuthUser authUser) {
+        // 해당 첨부파일 id값으로 존재하는지 검사
+        Attachment attachment = attachmentRepository.findByIdOrElseThrow(attachementId);
+        // 어떤 table(유저, 이벤트, 리뷰)에 대한 건지 검사 (tableRole 사용)
+        if (tableRole.equals(TableRole.EVENT)) {
+            // 해당 접근자가 작성한 table의 id값과 authUser값과 일치하는 해당 table의 객체가 존재하는지 검사
+            Event event = eventRepository.findByIdAndUserIdOrElseThrow(tableId, authUser.getUserId());
+            if(attachment.getTargetId().equals(event.getId())) {
+                String fileName = attachment.getUrl().substring(attachment.getUrl().lastIndexOf("/") + 1);
+                amazonS3Client.deleteObject(bucket, fileName);
+                attachmentRepository.delete(attachment);
+            }
+            
+        }
+        if (tableRole.equals(TableRole.USER)) {
+            // 해당 접근자가 작성한 table의 id값과 authUser값과 일치하는 해당 table의 객체가 존재하는지 검사
+            User user = userRepository.findByIdOrElseThrow(tableId);
+            if(attachment.getTargetId().equals(user.getId())) {
+                String fileName = attachment.getUrl().substring(attachment.getUrl().lastIndexOf("/") + 1);
+                amazonS3Client.deleteObject(bucket, fileName);
+                attachmentRepository.delete(attachment);
+            }
+        }
+        if (tableRole.equals(TableRole.REVIEW)) {
+            // 해당 접근자가 작성한 table의 id값과 authUser값과 일치하는 해당 table의 객체가 존재하는지 검사
+            // Review 나중에 합친거 보고 추후 추가 예정
+        }
```

Comment:
```suggestion
        if (tableRole.equals(TableRole.REVIEW)) {
            // TODO: 해당 접근자가 작성한 table의 id값과 authUser값과 일치하는 해당 table의 객체가 존재하는지 검사
            // TODO: Review 나중에 합친거 보고 추후 추가 예정
        }
```
나중에 추가하는 부분은 TODO 를 이용하면 인텔리제이의 도움을 받을 수 있습니다.